src/ImageSharp/Advanced/ParallelUtils/ParallelHelper.cs

```diff
@@ -62,14 +62,23 @@ public static void IterateRows(Rectangle rectangle, Configuration configuration,
 
             var parallelOptions = new ParallelOptions { MaxDegreeOfParallelism = numOfSteps };
 
+            int top = rectangle.Top;
+            int bottom = rectangle.Bottom;
+
             Parallel.For(
                 0,
                 numOfSteps,
                 parallelOptions,
                 i =>
                     {
-                        int yMin = rectangle.Top + (i * verticalStep);
-                        int yMax = Math.Min(yMin + verticalStep, rectangle.Bottom);
+                        int yMin = top + (i * verticalStep);
+
+                        if (yMin >= bottom)
+                        {
+                            return;
```

Comment:
Note to ourselves:
we launch a few threads out to the void, but it doesn't worth to bother more with this special case.

Reply:
Yeah, I did notice this. That's why I was hoping you were gonna add more tests, it took me a while to figure it all out. 😄